### PR TITLE
Updated ScaleIO Plugin

### PR DESCRIPTION
**Pull request by king-jam**
* Fixed Dynamic Namespaces not properly being created
* Updated to use plugin lib
* Updated example to be more current
* Refactored Storage Pool code to be more extensible

Fixes #

Summary of changes:
* Fixed Dynamic Namespaces not properly being created
* Updated to use plugin lib
* Updated example to be more current
* Refactored Storage Pool code to be more extensible

How to verify it:
Test Plugin capability using the example

Testing done:
Tested against real ScaleIO instance to verify functionality

A picture of a snapping turtle (not required but encouraged):
None that merits the honor

### CODE REVIEW

File: scaleio/pool.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		var metrics map[string]interface{}
+		err := s.getAPIResponse(fmt.Sprintf(statisticsPath, id), &metrics)
+		if err != nil {
+			return nil, err
+		}
+		for _, ns := range nss {
+			// Slice out only the important part for now
+			dyn := make([]plugin.NamespaceElement, len(ns))
+			copy(dyn, ns)
+			dyn[3].Value = id
```
Comment (sandlbn) [inline scaleio/pool.go]: can you add const instead of nr ?

**Review by sandlbn — CHANGES_REQUESTED** [review]
Looks great. If you can replace hardcoded number will be great.

**Comment by king-jam** [comment]
Removed the hardcoded numbers :+1:

**Review by sandlbn — APPROVED** [review]
LGTM

**Pull request merged**